faststreams/buffers.nim

```diff
@@ -335,7 +335,7 @@ template writeByte*(span: var PageSpan, val: byte) =
 
 template charsToBytes*(chars: openArray[char]): untyped =
   bind makeOpenArray
-  var charsStart = unsafeAddr chars[0]
```

Comment:
`baseAddr` has been fixed to return `ptr T` (so the `ptr byte` cast is no longer necessary) and to return nil when the source is empty, so this code will now be as safe as it gets